pennylane/beta/plugins/default_tensor.py

```diff
@@ -400,36 +443,37 @@ def _add_node(self, A, wires, name="UnnamedNode"):
             node = A
         else:
             node = tn.Node(A, name=name, backend=self.backend)
-        self._nodes.append(node)
+
+        if key not in self._nodes:
+            self._nodes[key] = []
+        self._nodes[key].append(node)
```

Comment:
I'm wondering if the bookkeeping can be avoided by making use of the built in `tensornetwork.quantum` module? C.f. https://github.com/google/TensorNetwork/blob/9f1f6ca753a906cc1c776f917b14beabe3910195/tensornetwork/quantum/quantum.py#L483

Reply:
I had a look. While we may want to consider it at some point, it doesn't seem to bypass the bookkeeping I'm doing here (which really only distinguishes three special cases: state/gate nodes, observable nodes, and a special node for the contracted pre-measurement state).